src/device.rs

```diff
@@ -547,25 +548,73 @@ impl<B: UsbBus> UsbDevice<'_, B> {
                 Ok(())
             }),
 
-            descriptor_type::STRING => {
-                if index == 0 {
-                    accept_writer(xfer, |w| {
-                        w.write(descriptor_type::STRING, &lang_id::ENGLISH_US.to_le_bytes())
-                    })
-                } else {
-                    let s = match index {
-                        1 => config.manufacturer,
-                        2 => config.product,
-                        3 => config.serial_number,
-                        _ => {
-                            let index = StringIndex::new(index);
-                            let lang_id = req.index;
+            descriptor_type::STRING => match index {
+                // first STRING Request
+                0 => {
+                    if let Some(extra_lang_ids) = config.extra_lang_ids {
+                        assert!(
+                            extra_lang_ids.len() < 16,
+                            "Not support more than 15 extra LangIDs"
+                        );
```

Comment:
This should be done when adding lang IDs to the config, not at run-time. Please move this assert elsewhere.

Reply:
I will remove this `assert!`, I think the `assert!` in the methods of `UsbDeviceBuilder` will be enough.

Reply:
done